lib/driver.js

```diff
@@ -123,6 +123,7 @@ class AndroidDriver extends BaseDriver {
 
       // get device udid for this session
       let {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
+      this.opts.avd = this.opts.avd || /emulator/.test(udid);
```

Comment:
so we are potentially setting `this.opts.avd` to `true` rather than an avd name? hmm  hmm. i wonder if it'd be more sensible to just track the udid in `this.opts.udid` or something like that, and then augment `isEmulator` to test for either case? just not sure that saturating `this.opts.avd` with more semantics is a good idea.